### PR TITLE
 Show a delete confirmation modal when deleting task orders

### PR DESCRIPTION
**Pull request by roseeichelmann**
## Associated issues
Fixes cityofaustin/atd-data-tech/issues/11069

Shoutout to @tillyw for the wonderful code I borrowed :)

## Testing
**URL to test:** 

https://deploy-preview-923--atd-moped-main.netlify.app/moped/projects

**Steps to test:**
1. Go to the funding page of a project
2. Add task orders
3. Upon trying to delete a task order you should get a delete confirmation modal

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectFundingTable.js [hunk 1]
```diff
@@ -586,15 +595,22 @@ const ProjectFundingTable = () => {
...
-                </li>
-              ))}
+              <DeleteConfirmationModal
+                type="tag"
+                submitDelete={() => handleTaskOrderDelete(deleteConfirmationId)}
+                isDeleteConfirmationOpen={isDeleteConfirmationOpen}
+                setIsDeleteConfirmationOpen={setIsDeleteConfirmationOpen}
+              >
+                {taskOrderData.map((task) => (
+                  <li key={task.id}>
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectFundingTable.js:605]: this issue pre-dates your work, but i'm seeing a unique key prop error here. the problem is that `task.id` is `undefined` for all of these records. you can switch this to `task.task_order` to clear i tup.

File: moped-editor/src/views/projects/projectView/ProjectFundingTable.js [hunk 2]
```diff
@@ -586,15 +595,22 @@ const ProjectFundingTable = () => {
             />
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectFundingTable.js:595]: just pointing out that that i was surprised to see this entire feature crammed into the `title` prop of material table 🤔

**Review by johnclary — CHANGES_REQUESTED** [review]
it's great to see this component re-used—and i'm so glad we have this delete protection in place. 

please fix up that unique key property but otherwise 🚢 💨  🙌

File: moped-editor/src/views/projects/projectView/ProjectFundingTable.js (hunk 2 above)
Comment (chiaberry) [reply moped-editor/src/views/projects/projectView/ProjectFundingTable.js:595]: 😬

**Comment by roseeichelmann** [comment]
@johnclary fixed up that unique key prop error!

**Review by chiaberry — APPROVED** [review]

**Review by johnclary — APPROVED** [review]
🚢 🚢 🚢 🚢 🚢 🚢

**Pull request merged**